Add minimum disk space requirements for the components (#8388)
pingcap/docs: hardware-and-software-requirements.md

| Alertmanager | 9093 | the port for the alert web service |
| Alertmanager | 9094 | the alert communication port |

## Disk space requirements

| Component | Disk space requirement |
| :-- | :-- |
| TiDB |  |

> | PD | Reserve at least 30 GB for data and log disks |

Author: ```suggestion
| PD | Reserve at least 20 GB for data and log disks |
```

> | TiDB | Reserve at least 30 GB for the log disk | Lower than 90% |

Reviewer: I don't think "Reserve" is necessary here.
```suggestion
| TiDB | At least 30 GB for the log disk | Lower than 90% |
```

> | PD | Reserve at least 20 GB for the data disk and the log disk, respectively | Lower than 90% |

Reviewer: ```suggestion
| PD | At least 20 GB for the data disk and for the log disk, respectively | Lower than 90% |
```

> | TiKV | Reserve at least 100 GB for the data disk and the log disk, respectively | Lower than 80% |

Reviewer: ditto

> | TiFlash | Reserve at least 100 GB and 30 GB for the data disk and log disk, respectively | Lower than 80% |

Reviewer 2: ```suggestion
| TiFlash | Reserve at least 100 GB and 30 GB for the data disk and the log disk, respectively | Lower than 80% |
```
Should there be a "the" before "log disk"?

> | TiFlash | Reserve at least 100 GB and 30 GB for the data disk and the log disk, respectively | Lower than 80% |

Reviewer: ```suggestion
| TiFlash | At least 100 GB for the data disk, and at least 30 GB for the log disk, respectively | Lower than 80% |
```

> | TiFlash | TiFlash binary: Occupies 2.1 GB. Considering log space, you are advised to reserve no less than 30 GB space for the deployment directory <br/> Data disk: No less than 100 GB is recommended. 20 GB is the minimum requirement. Ensure that the disk usage is below 80% |

Author: ```suggestion
| TiFlash | Log disk: Reserve at least 30 GB <br/> Data disk: Reserve at least 100 GB. Ensure that the disk usage is below 80% |
```

> | TiUP | A TiUP cluster occupies no more than 1 GB. During deployment, the TiUP cluster requires no more than 1 MB /tmp space to restore temporary configuration files | N/A |

Reviewer: 见 https://github.com/pingcap/docs-cn/pull/9202/files#r867659972
```suggestion
| TiUP | A single TiUP cluster occupies no more than 1 GB. During the cluster deployment process, the TiUP cluster requires no more than 1 MB /tmp space to restore temporary configuration files. | N/A |
```

>  Ngmonitoring | Conprof: 3 x 1 GB x Number of components + 20 GB (reserved space. 3 days, 1 GB for each component per day)<br/>TopSQL: 30 x 50 MB x Number of components (30 days, 50 MB for each component per day, shared with Continuous Profiling) |

Reviewer 2: ```suggestion
| Ngmonitoring | Conprof: 3 x 1 GB x Number of components + 20 GB (reserved space. 3 days, 1 GB for each component per day)<br/>TopSQL: 30 x 50 MB x Number of components (30 days, 50 MB for each component per day, shared with Continuous Profiling) |
```

> | Ngmonitoring | Conprof: 3 x 1 GB x Number of components (3 days, each component occupies about 1 GB per day) + 20 GB reserved space <br/> Top SQL: 30 x 50 MB x Number of components (30 days, each component occupies about 50 MB per day) <br/> Conprof and Top SQL share the reserved space | N/A |

Reviewer: ```suggestion
| Ngmonitoring | Conprof: 3 x 1 GB x Number of components (which means each component occupies about 1 GB per day in a total of 3 days) + 20 GB reserved space <br/> Top SQL: 30 x 50 MB x Number of components (which means each component occupies about 50 MB per day in a total of 30 days) <br/> Conprof and Top SQL share the reserved space | N/A |
```

> | TiUP | <ul><li>Control machine: No more than 1 GB space is required for deploying a single version of TiDB cluster. The space required increases if multiple versions of TiDB cluster are deployed. </li> <li> Deployment servers (machines where the TiDB components run): TiFlash occupies about 700 MB space and other components (such as PD, TiDB, and TiKV) occupy about 200 MB space respectively. During the cluster deployment process, the TiUP cluster requires less than 1 MB temporary space (`/tmp` directory) to store temporary files.</li></ul>| N/A |

Reviewer: The Chinese version says "其他组件（PD、TiDB、TiKV 等）占用约 200 MB 空间" WITHOUT mentioning "respectively". This is inconsistent.
```suggestion
| TiUP | <ul><li>Control machine: No more than 1 GB space is required for deploying a TiDB cluster of a single version. The space required increases if TiDB clusters of multiple versions are deployed. </li> <li> Deployment servers (machines where the TiDB components run): TiFlash occupies about 700 MB space and other components (such as PD, TiDB, and TiKV) occupy about 200 MB space respectively. During the cluster deployment process, the TiUP cluster requires less than 1 MB of temporary space (`/tmp` directory) to store temporary files.</li></ul>| N/A |
```
